### ISSUE TITLE
[Swift 4.1] Add configuration option to fire `explicit_type_interface` only for properties rather than local variables

### ISSUE THREAD

**Issue by rounak** [realm/SwiftLint#2028]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.24.2
* Installation method used (Homebrew, CocoaPods, building from source, etc)? portable
* Paste your configuration file:

* Which Xcode version are you using (check `xcode-select -p`)? 9.3

```swift
// This triggers a violation:
func a() {
    let foo = UIView()
}
```

With reference to issue #2027

**Comment by marcelofabri** [comment]
Fixed in #2060 🎉
